src/util/JoinAlgorithms/JoinAlgorithms.h

```diff
@@ -40,6 +40,13 @@ CPP_concept BinaryIteratorFunction =
     std::invocable<F, ql::ranges::iterator_t<Range>,
                    ql::ranges::iterator_t<Range>>;
 
+// Helper type to indicate the different join modes.
+enum class JoinType { JOIN, OPTIONAL, MINUS };
+
+// Helper type to indicate the mode at compile-time.
+template <JoinType Mode>
+using ConstantJoinType = std::integral_constant<JoinType, Mode>;
```

Comment:
make static constexpr constants for these things.

`MinusJoinTag` `OptionalJoinTag` `OrdinaryJoinTag`.

Reply:
(makes the call site much simpler)